test/SILOptimizer/infinite_recursion.swift

```diff
@@ -0,0 +1,121 @@
+// RUN: %target-swift-frontend -emit-sil -primary-file %s -o /dev/null -verify
+
+func a() {  // expected-warning {{all paths through this function will call itself}}
+  a()
+}
+
+func b(_ x : Int) {  // expected-warning {{all paths through this function will call itself}}
+  if x != 0 {
+    b(x)
+  } else {
+    b(x+1)
+  }
+}
+
+func c(_ x : Int) {
+  if x != 0 {
+    c(5)
+  }
+}
+
+func d(_ x : Int) {  // expected-warning {{all paths through this function will call itself}}
+  var x = x
+  if x != 0 {
+    x += 1
+  }
+  return d(x)
+}
+
+// Doesn't warn on mutually recursive functions
+
+func e() { f() }
+func f() { e() }
+
+class C {
+  lazy var a: String = {
+    let a = "test"
+    print(self.a) // Should warn - interprocedural cycle.
```

Comment:
This is SR-5224 :)

Reply:
I think I know how to fix it too!  Standby

Reply:
<3